manifests/resource/server.pp

```diff
@@ -316,9 +320,39 @@
     }
   }
 
+  # If port range is defined, ignore any other $listen_port defined
+  if versioncmp(fact('nginx_version'), '1.15.10') < 0 {
+    $port_range_support = false
+  }
+  else{
+    $port_range_support = true
+  }
+
+  if ($listen_port_range != undef) and ($port_range_support == true) {
+    $port = $listen_port_range
+  }
+  elsif ($listen_port_range != undef) and ($port_range_support == false) {
+    fail('nginx: this version of nginx does not support port ranges (must be >= 1.15.10)')
+  }
+  else{
+    $port = $listen_port
+  }
+
+  if $ipv6_enable == true{
+    if ($ipv6_listen_port_range != undef) and ($port_range_support == true) {
+      $ipv6_port = $ipv6_listen_port_range
+    }
+    elsif ($ipv6_listen_port_range != undef) and ($port_range_support == false) {
+      fail('nginx: this version of nginx does not support port ranges (must be >= 1.15.10)')
+    }
+    else {
+      $ipv6_port = $ipv6_listen_port
+    }
+  }
+
```

Comment:
@00ssh I think your code is to complicated and does not need to throw an error message. Maybe something like this could work:
```
  # If port range is defined, ignore any other $listen_port defined
  if versioncmp(fact('nginx_version'), '1.15.10') < 0 {
    $port = $listen_port

    if $ipv6_enable == true {
      $ipv6_port = $ipv6_listen_port
    }
  } else {
    $port = pick_default($listen_port_range, $listen_port)

    if $ipv6_enable == true {
      $ipv6_port = pick_default($ipv6_listen_port_range, $ipv6_listen_port)
    }
  }
```
https://github.com/puppetlabs/puppetlabs-stdlib/blob/master/REFERENCE.md#pick_default

Reply:
@dhoppe Hi, Thanks for your suggestions. I've just tested with pick_default and it works just fine. However, I still have to figure out why the travis pipeline is failing (I don't have an environment connected to the internet right now). It looks like "fact('nginx_version')" doesn't return anything.

Reply:
@00ssh Maybe, for that reason I used ```$facts['nginx_version']``` at the example from above.

Reply:
@dhoppe Thanks. I've updated accordingly. Waiting for tests to pass this time!